### PR TITLE
feat: support custom workspaceRoot for angular CT

### PR DESCRIPTION
**Pull request by barbados-clemens**
Nx issue for ref: https://github.com/nrwl/nx/issues/12113

### Additional details

This is the same implementation as https://github.com/cypress-io/cypress/pull/25384 but to solve a different issue, primarly the issue where Nx + Cypress Angular CT is unable to load the workspace root tailwinds config file. This logic is hard coded within angular webpack utils that cypress CT calls.
```ts
function getTailwindConfigPath({ projectRoot, root }: WebpackConfigOptions): string | undefined {
  // A configuration file can exist in the project or workspace root
  // The list of valid config files can be found:
  // https://github.com/tailwindlabs/tailwindcss/blob/8845d112fb62d79815b50b3bae80c317450b8b92/src/util/resolveConfigPath.js#L46-L52
  const tailwindConfigFiles = ['tailwind.config.js', 'tailwind.config.cjs'];
  for (const basePath of [projectRoot, root]) {
    for (const configFile of tailwindConfigFiles) {
      // Irrespective of the name project level configuration should always take precedence.
      const fullPath = path.join(basePath, configFile);
      if (fs.existsSync(fullPath)) {
        return fullPath;
      }
    }
  }

  return undefined;
}
```

The Angular CT handler sets the projectRoot as the workspaceRoot when working with the angular webpack utils which works when cypress is opened in the root of the workspace. But within an nx context Cypress is opened in the project root which means any tailwind config files in the workspace root aren't loaded. 

This PR allows for an optional workspaceRoot option to be passed via the cypress config component testing project configuration. if one isn't provided then the default value of projectRoot is used to prevent any other integration/assumptions from breaking but allowing the customizations for tools like Nx to provide the expected behavior when the Cypress isn't opened in the workspaceRoot. 

### Steps to test

This change does requires changes to the @nrwl/angular nx component testing preset. Mainly the offset value is removed and a custom 'workspaceRoot' is passed to cypress. This can be done within node_modules if really wanted to test it works before this change is in Nx, See: https://github.com/nrwl/nx/pull/15485

make a new nx angular workspace, npx create-nx-workspace@latest --preset=angular
add an tailwind to the workspace root
use a tailwind class in the apps create 'apps/<app-name>/src/app.component.html'
add CT to the app `nx g @nrwl/angular:cypress-component-configuration --project=<app-name>`
run CT and notice the tailwind styles are not applied, `nx component-test <app-name> --watch`

cloneable repo: https://github.com/kanidjar/nx-cypress-ct-styling-bug

### How has the user experience changed?

No impact for direct consumers, more flexibility for 3rd party tools like [Nx](https://nx.dev)

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by jordanpowell88** [comment]
I think this looks good to me @barbados-clemens. The only thing I think we need to add are `system-tests` using Nx. This will ensure we don't break features like this moving forward and that we can continue to validate it works with Nx.

**Review by jordanpowell88 — APPROVED** [review]
LGTM - I want to try to add a few Nx system tests first though before merging

**Comment by lmiller1990** [comment]
Looks like the nx test *is* failing. Once this is passing, we can merge it up. Are you interested in picking this one up @jordanpowell88 to help get over the line?

We've also got another issue filed for adding more robust Nx infra to the monorepo.

**Comment by mike-plummer** [comment]
@jordanpowell88 Are you looking at the nx failure on this one?

**Comment by jordanpowell88** [comment]
@mike-plummer It's blocked until we do this issue: https://github.com/cypress-io/cypress/issues/26045

File: npm/webpack-dev-server/src/helpers/angularHandler.ts
Comment (mike-plummer) [inline npm/webpack-dev-server/src/helpers/angularHandler.ts:257]: Could this just use the `buildOptions` output from `getAngularBuildOptions` above (same on line 279)? Slightly cleaner, perhaps

**Review by mike-plummer — APPROVED** [review]
Looks good, just one stylistic comment

**Pull request merged**
